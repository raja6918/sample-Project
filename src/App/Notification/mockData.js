export const mockData = {
  data: [
    {
      activeTime: '0:00:05',
      alertType: 'NOTICE',
      creationDateTime: '2020-08-12T07:57:15+0000',
      elapsedTime: '0:00:00',
      eventId: 'JOB1001',
      eventType: 'solver',
      jobId: 'JOB1001',
      jobOwner: 11,
      lastModified: '2020-08-12T07:57:15+0000',
      message: 'string',
      notificationType: 'solver',
      scenarioId: 2,
      scenarioName: 'Test',
      solverId: 1,
      solverName: 'Test',
      state: 'Queued',
      status: 'Done-success',
      viewStatus: 'SEEN',
    },
  ],
  endIndex: 20,
  notViewedCount: 20,
  startIndex: 0,
  totalCount: 40,
};
