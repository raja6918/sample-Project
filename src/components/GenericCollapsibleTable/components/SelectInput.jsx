import React, { Component, createRef } from 'react';
import PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import ResetButton from './ResetButton';
import { ERROR_ZINDEX } from './constants';
import './style.scss';

export default class SelectInput extends Component {
  state = {
    selectValue:
      this.props.value === null ? this.props.defaultValue : this.props.value,
  };

  selector = createRef();

  componentWillReceiveProps(nextProps) {
    // To revert to previous value if the API call failed if value props is injected
    if (nextProps.value !== null) {
      if (nextProps.value !== this.state.selectValue && !nextProps.error) {
        this.setState({
          selectValue: nextProps.value,
        });
      }
    }
  }

  handleChange = e => {
    this.setState({ selectValue: e.target.value }, () => {
      this.props.onChange(e.target.value, this.props.data);
    });
    // To remove input focus after selection
    this.selector.current.children[0].style.background = 'transparent';
    this.selector.current.classList.remove('Mui-focused');
  };

  handleClick = () => {
    // To remove input focus after click
    this.selector.current.children[0].style.background = 'transparent';
    this.selector.current.classList.remove('Mui-focused');

    // To remove any outer overlay if any when we use this component
    this.props.removeOverlay();
  };

  generateStyle = () => {
    const zIndex = this.props.error ? ERROR_ZINDEX : 1;
    return { zIndex, ...this.props.style };
  };

  resetToDefaultValue = () => {
    this.setState({
      selectValue:
        this.props.value === undefined || this.props.value === null
          ? this.props.defaultValue
          : this.props.value,
    });
  };

  render() {
    const {
      value,
      data,
      items,
      error,
      handleDisable,
      enableReset,
      handleReset,
      handleTooltipDisable,
      getTooltipContent,
      placeholder,
      defaultValue,
    } = this.props;
    const { selectValue } = this.state;

    const disabled = handleDisable(value, data);
    const tooltipDisabled = handleTooltipDisable(value, data);
    const tooltipContent = getTooltipContent(value, data);

    return (
      <span className={error ? 'input-error' : ''}>
        {' '}
        <Select
          style={this.generateStyle()}
          onChange={this.handleChange}
          value={selectValue}
          disabled={disabled}
          ref={this.selector}
          onClick={this.handleClick}
          className={
            selectValue === defaultValue ? 'select-input-placeholder' : ''
          }
        >
          <MenuItem value={defaultValue} style={{ display: 'none' }} disabled>
            {placeholder}
          </MenuItem>
          {items.map((item, k) => (
            <MenuItem key={k} value={item.value}>
              {item.display}
            </MenuItem>
          ))}
        </Select>
        {enableReset && (
          <span>
            (
            <ResetButton
              data={data}
              handleReset={handleReset}
              disabled={tooltipDisabled}
              tooltipContent={tooltipContent}
            />
            )
          </span>
        )}{' '}
      </span>
    );
  }
}

SelectInput.propTypes = {
  data: PropTypes.shape(),
  items: PropTypes.arrayOf(
    PropTypes.shape({
      display: PropTypes.string,
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.string,
        PropTypes.number,
      ]),
    })
  ).isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
  ]),
  style: PropTypes.shape(),
  error: PropTypes.bool,
  handleDisable: PropTypes.func,
  enableReset: PropTypes.bool,
  handleReset: PropTypes.func,
  handleTooltipDisable: PropTypes.func,
  getTooltipContent: PropTypes.func,
  removeOverlay: PropTypes.func,
  defaultValue: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
    PropTypes.null,
  ]),
  placeholder: PropTypes.string,
};

SelectInput.defaultProps = {
  style: { width: '100%' },
  error: false,
  handleDisable: () => false,
  enableReset: false,
  handleReset: () => null,
  handleTooltipDisable: () => false,
  getTooltipContent: () => '',
  removeOverlay: () => null,
  defaultValue: '_default',
  data: {},
  placeholder: '',
  value: null,
};
