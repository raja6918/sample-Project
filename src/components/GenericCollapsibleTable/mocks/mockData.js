import { ellipsisTransformer } from '../../GenericTable/transformers';
import { SelectInput, ColorHighlighter } from '../components';

export const stateData = [
  {
    value: true,
    display: 'On',
  },
  {
    value: false,
    display: 'Off',
  },
];

const handleChange = value => {
  console.log('handle SelectInput Change', value);
};

export const getHeaders = t => [
  {
    field: 'name',
    displayName: t('DATA.rules.headers.name'),
    transformer: ellipsisTransformer,
    transformerStyle: {
      maxWidth: '235px',
    },
  },
  {
    field: 'state',
    displayName: t('DATA.rules.headers.state'),
    disableFilter: true,
    component: SelectInput,
    props: {
      onChange: handleChange,
      items: stateData,
    },
    componentStyle: {
      width: '120px',
      fontSize: '0.8125rem',
    },
    cellStyle: {
      width: '120px',
    },
  },
  {
    field: 'settingsFrom',
    displayName: t('DATA.rules.headers.settingsFrom'),
    component: ColorHighlighter,
    componentStyle: {
      color: '#FF650C',
    },
    props: {
      header: 'isEdited',
    },
  },
  {
    field: 'catagory',
    displayName: t('DATA.rules.headers.catagory'),
  },
  {
    field: 'type',
    displayName: t('DATA.rules.headers.type'),
  },
];

export const rulesData = [
  {
    state: true,
    settingsFrom: 'Baseline Rules',
    catagory: 'CAR Government Regulation',
    name: 'Night Duty',
    type: 'Definition',
    id: 1,
  },
  {
    state: false,
    settingsFrom: 'Flight Desk',
    catagory: 'CAR Government Regulation',
    name:
      'Split Duty Extension to Maximum FDP (Split Duty Extension to Maximum FDP)',
    type: 'Duty Time Limitation',
    id: 2,
  },
  {
    state: true,
    settingsFrom: 'Baseline Rules',
    catagory: 'CAR Government Regulation',
    name: 'Minimum Rest at Home Base (Minimum Rest at Home Base)',
    type: 'Rest RUle',
    id: 3,
  },
  {
    isEdited: true,
    state: false,
    settingsFrom: 'Widebody',
    catagory: 'OverSeas Cabin Contract',
    name: 'Earily Duty',
    type: 'Cost RUle',
    id: 4,
  },
  {
    state: true,
    settingsFrom: 'Baseline Rules',
    catagory: 'Basic Pairing Rules',
    name: 'Late Duty',
    type: 'Soft Rule',
    id: 5,
  },
];
