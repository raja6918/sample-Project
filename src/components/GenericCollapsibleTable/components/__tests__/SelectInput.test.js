import React from 'react';
import { mount, shallow } from 'enzyme';
import SelectInput from '../SelectInput';
import { stateData } from '../../mocks/mockData';

const mockProps = {
  data: { name: 'hello' },
  items: stateData,
  onChange: value => value,
  value: true,
  style: { width: '100%' },
  handleDisable: () => false,
  removeOverlay: jest.fn(),
};

let wrapper;
beforeAll(() => {
  wrapper = mount(<SelectInput {...mockProps} />);
});

afterAll(() => {
  wrapper.unmount();
});

test('SelectInput Component render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('check whether selectValue state value changed on onChange event', () => {
  expect(wrapper.state('selectValue')).toEqual(true);

  wrapper.find('WithStyles(ForwardRef(Select))').prop('onChange')({
    target: {
      value: false,
    },
  });

  expect(wrapper.state('selectValue')).toEqual(false);
});

test('check whether removeOverlay prop is called on click event', () => {
  wrapper.find('WithStyles(ForwardRef(Select))').prop('onClick')();

  expect(mockProps.removeOverlay).toBeCalled();
});

let readOnlyWrapper;
test('SelectInput Component render correctly when it is disabled', () => {
  readOnlyWrapper = shallow(
    <SelectInput {...mockProps} handleDisable={() => true} />
  );
  expect(readOnlyWrapper).toMatchSnapshot();
});
