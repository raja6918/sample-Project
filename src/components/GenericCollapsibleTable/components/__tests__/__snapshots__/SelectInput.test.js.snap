// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput Component render correctly 1`] = `
<SelectInput
  data={
    Object {
      "name": "hello",
    }
  }
  defaultValue="_default"
  enableReset={false}
  error={false}
  getTooltipContent={[Function]}
  handleDisable={[Function]}
  handleReset={[Function]}
  handleTooltipDisable={[Function]}
  items={
    Array [
      Object {
        "display": "On",
        "value": true,
      },
      Object {
        "display": "Off",
        "value": false,
      },
    ]
  }
  onChange={[Function]}
  placeholder=""
  removeOverlay={[MockFunction]}
  style={
    Object {
      "width": "100%",
    }
  }
  value={true}
>
  <span
    className=""
  >
     
    <WithStyles(ForwardRef(Select))
      className=""
      disabled={false}
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "width": "100%",
          "zIndex": 1,
        }
      }
      value={true}
    >
      <ForwardRef(Select)
        className=""
        classes={
          Object {
            "disabled": "Mui-disabled",
            "filled": "MuiSelect-filled",
            "icon": "MuiSelect-icon",
            "iconFilled": "MuiSelect-iconFilled",
            "iconOpen": "MuiSelect-iconOpen",
            "iconOutlined": "MuiSelect-iconOutlined",
            "nativeInput": "MuiSelect-nativeInput",
            "outlined": "MuiSelect-outlined",
            "root": "MuiSelect-root",
            "select": "MuiSelect-select",
            "selectMenu": "MuiSelect-selectMenu",
          }
        }
        disabled={false}
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "width": "100%",
            "zIndex": 1,
          }
        }
        value={true}
      >
        <WithStyles(ForwardRef(Input))
          className=""
          disabled={false}
          inputComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "IconComponent": [Function],
                "MenuProps": [Function],
                "SelectDisplayProps": [Function],
                "aria-label": [Function],
                "autoFocus": [Function],
                "autoWidth": [Function],
                "children": [Function],
                "className": [Function],
                "classes": [Function],
                "defaultValue": [Function],
                "disabled": [Function],
                "displayEmpty": [Function],
                "inputRef": [Function],
                "labelId": [Function],
                "multiple": [Function],
                "name": [Function],
                "onBlur": [Function],
                "onChange": [Function],
                "onClose": [Function],
                "onFocus": [Function],
                "onOpen": [Function],
                "open": [Function],
                "readOnly": [Function],
                "renderValue": [Function],
                "tabIndex": [Function],
                "type": [Function],
                "value": [Function],
                "variant": [Function],
              },
              "render": [Function],
            }
          }
          inputProps={
            Object {
              "IconComponent": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "MenuProps": undefined,
              "SelectDisplayProps": Object {
                "id": undefined,
              },
              "autoWidth": false,
              "children": Array [
                <WithStyles(ForwardRef(MenuItem))
                  disabled={true}
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                  value="_default"
                >
                  
                </WithStyles(ForwardRef(MenuItem))>,
                Array [
                  <WithStyles(ForwardRef(MenuItem))
                    value={true}
                  >
                    On
                  </WithStyles(ForwardRef(MenuItem))>,
                  <WithStyles(ForwardRef(MenuItem))
                    value={false}
                  >
                    Off
                  </WithStyles(ForwardRef(MenuItem))>,
                ],
              ],
              "classes": Object {
                "disabled": "Mui-disabled",
                "filled": "MuiSelect-filled",
                "icon": "MuiSelect-icon",
                "iconFilled": "MuiSelect-iconFilled",
                "iconOpen": "MuiSelect-iconOpen",
                "iconOutlined": "MuiSelect-iconOutlined",
                "nativeInput": "MuiSelect-nativeInput",
                "outlined": "MuiSelect-outlined",
                "root": "MuiSelect-root",
                "select": "MuiSelect-select",
                "selectMenu": "MuiSelect-selectMenu",
              },
              "displayEmpty": false,
              "labelId": undefined,
              "multiple": false,
              "onClose": undefined,
              "onOpen": undefined,
              "open": undefined,
              "renderValue": undefined,
              "type": undefined,
              "variant": "standard",
            }
          }
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
              "zIndex": 1,
            }
          }
          value={true}
        >
          <ForwardRef(Input)
            className=""
            classes={
              Object {
                "colorSecondary": "MuiInput-colorSecondary",
                "disabled": "Mui-disabled",
                "error": "Mui-error",
                "focused": "Mui-focused",
                "formControl": "MuiInput-formControl",
                "fullWidth": "MuiInput-fullWidth",
                "input": "MuiInput-input",
                "inputMarginDense": "MuiInput-inputMarginDense",
                "inputMultiline": "MuiInput-inputMultiline",
                "inputTypeSearch": "MuiInput-inputTypeSearch",
                "marginDense": "MuiInput-marginDense",
                "multiline": "MuiInput-multiline",
                "root": "MuiInput-root",
                "underline": "MuiInput-underline",
              }
            }
            disabled={false}
            inputComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "IconComponent": [Function],
                  "MenuProps": [Function],
                  "SelectDisplayProps": [Function],
                  "aria-label": [Function],
                  "autoFocus": [Function],
                  "autoWidth": [Function],
                  "children": [Function],
                  "className": [Function],
                  "classes": [Function],
                  "defaultValue": [Function],
                  "disabled": [Function],
                  "displayEmpty": [Function],
                  "inputRef": [Function],
                  "labelId": [Function],
                  "multiple": [Function],
                  "name": [Function],
                  "onBlur": [Function],
                  "onChange": [Function],
                  "onClose": [Function],
                  "onFocus": [Function],
                  "onOpen": [Function],
                  "open": [Function],
                  "readOnly": [Function],
                  "renderValue": [Function],
                  "tabIndex": [Function],
                  "type": [Function],
                  "value": [Function],
                  "variant": [Function],
                },
                "render": [Function],
              }
            }
            inputProps={
              Object {
                "IconComponent": Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "MenuProps": undefined,
                "SelectDisplayProps": Object {
                  "id": undefined,
                },
                "autoWidth": false,
                "children": Array [
                  <WithStyles(ForwardRef(MenuItem))
                    disabled={true}
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    value="_default"
                  >
                    
                  </WithStyles(ForwardRef(MenuItem))>,
                  Array [
                    <WithStyles(ForwardRef(MenuItem))
                      value={true}
                    >
                      On
                    </WithStyles(ForwardRef(MenuItem))>,
                    <WithStyles(ForwardRef(MenuItem))
                      value={false}
                    >
                      Off
                    </WithStyles(ForwardRef(MenuItem))>,
                  ],
                ],
                "classes": Object {
                  "disabled": "Mui-disabled",
                  "filled": "MuiSelect-filled",
                  "icon": "MuiSelect-icon",
                  "iconFilled": "MuiSelect-iconFilled",
                  "iconOpen": "MuiSelect-iconOpen",
                  "iconOutlined": "MuiSelect-iconOutlined",
                  "nativeInput": "MuiSelect-nativeInput",
                  "outlined": "MuiSelect-outlined",
                  "root": "MuiSelect-root",
                  "select": "MuiSelect-select",
                  "selectMenu": "MuiSelect-selectMenu",
                },
                "displayEmpty": false,
                "labelId": undefined,
                "multiple": false,
                "onClose": undefined,
                "onOpen": undefined,
                "open": undefined,
                "renderValue": undefined,
                "type": undefined,
                "variant": "standard",
              }
            }
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "width": "100%",
                "zIndex": 1,
              }
            }
            value={true}
          >
            <WithStyles(ForwardRef(InputBase))
              className=""
              classes={
                Object {
                  "colorSecondary": "MuiInput-colorSecondary",
                  "disabled": "Mui-disabled",
                  "error": "Mui-error",
                  "focused": "Mui-focused",
                  "formControl": "MuiInput-formControl",
                  "fullWidth": "MuiInput-fullWidth",
                  "input": "MuiInput-input",
                  "inputMarginDense": "MuiInput-inputMarginDense",
                  "inputMultiline": "MuiInput-inputMultiline",
                  "inputTypeSearch": "MuiInput-inputTypeSearch",
                  "marginDense": "MuiInput-marginDense",
                  "multiline": "MuiInput-multiline",
                  "root": "MuiInput-root MuiInput-underline",
                  "underline": null,
                }
              }
              disabled={false}
              fullWidth={false}
              inputComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "IconComponent": [Function],
                    "MenuProps": [Function],
                    "SelectDisplayProps": [Function],
                    "aria-label": [Function],
                    "autoFocus": [Function],
                    "autoWidth": [Function],
                    "children": [Function],
                    "className": [Function],
                    "classes": [Function],
                    "defaultValue": [Function],
                    "disabled": [Function],
                    "displayEmpty": [Function],
                    "inputRef": [Function],
                    "labelId": [Function],
                    "multiple": [Function],
                    "name": [Function],
                    "onBlur": [Function],
                    "onChange": [Function],
                    "onClose": [Function],
                    "onFocus": [Function],
                    "onOpen": [Function],
                    "open": [Function],
                    "readOnly": [Function],
                    "renderValue": [Function],
                    "tabIndex": [Function],
                    "type": [Function],
                    "value": [Function],
                    "variant": [Function],
                  },
                  "render": [Function],
                }
              }
              inputProps={
                Object {
                  "IconComponent": Object {
                    "$$typeof": Symbol(react.memo),
                    "compare": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  },
                  "MenuProps": undefined,
                  "SelectDisplayProps": Object {
                    "id": undefined,
                  },
                  "autoWidth": false,
                  "children": Array [
                    <WithStyles(ForwardRef(MenuItem))
                      disabled={true}
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                      value="_default"
                    >
                      
                    </WithStyles(ForwardRef(MenuItem))>,
                    Array [
                      <WithStyles(ForwardRef(MenuItem))
                        value={true}
                      >
                        On
                      </WithStyles(ForwardRef(MenuItem))>,
                      <WithStyles(ForwardRef(MenuItem))
                        value={false}
                      >
                        Off
                      </WithStyles(ForwardRef(MenuItem))>,
                    ],
                  ],
                  "classes": Object {
                    "disabled": "Mui-disabled",
                    "filled": "MuiSelect-filled",
                    "icon": "MuiSelect-icon",
                    "iconFilled": "MuiSelect-iconFilled",
                    "iconOpen": "MuiSelect-iconOpen",
                    "iconOutlined": "MuiSelect-iconOutlined",
                    "nativeInput": "MuiSelect-nativeInput",
                    "outlined": "MuiSelect-outlined",
                    "root": "MuiSelect-root",
                    "select": "MuiSelect-select",
                    "selectMenu": "MuiSelect-selectMenu",
                  },
                  "displayEmpty": false,
                  "labelId": undefined,
                  "multiple": false,
                  "onClose": undefined,
                  "onOpen": undefined,
                  "open": undefined,
                  "renderValue": undefined,
                  "type": undefined,
                  "variant": "standard",
                }
              }
              multiline={false}
              onChange={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "width": "100%",
                  "zIndex": 1,
                }
              }
              type="text"
              value={true}
            >
              <ForwardRef(InputBase)
                className=""
                classes={
                  Object {
                    "adornedEnd": "MuiInputBase-adornedEnd",
                    "adornedStart": "MuiInputBase-adornedStart",
                    "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                    "disabled": "Mui-disabled Mui-disabled",
                    "error": "Mui-error Mui-error",
                    "focused": "Mui-focused Mui-focused",
                    "formControl": "MuiInputBase-formControl MuiInput-formControl",
                    "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                    "input": "MuiInputBase-input MuiInput-input",
                    "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                    "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                    "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                    "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                    "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                    "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                    "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                    "multiline": "MuiInputBase-multiline MuiInput-multiline",
                    "root": "MuiInputBase-root MuiInput-root MuiInput-underline",
                  }
                }
                disabled={false}
                fullWidth={false}
                inputComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "propTypes": Object {
                      "IconComponent": [Function],
                      "MenuProps": [Function],
                      "SelectDisplayProps": [Function],
                      "aria-label": [Function],
                      "autoFocus": [Function],
                      "autoWidth": [Function],
                      "children": [Function],
                      "className": [Function],
                      "classes": [Function],
                      "defaultValue": [Function],
                      "disabled": [Function],
                      "displayEmpty": [Function],
                      "inputRef": [Function],
                      "labelId": [Function],
                      "multiple": [Function],
                      "name": [Function],
                      "onBlur": [Function],
                      "onChange": [Function],
                      "onClose": [Function],
                      "onFocus": [Function],
                      "onOpen": [Function],
                      "open": [Function],
                      "readOnly": [Function],
                      "renderValue": [Function],
                      "tabIndex": [Function],
                      "type": [Function],
                      "value": [Function],
                      "variant": [Function],
                    },
                    "render": [Function],
                  }
                }
                inputProps={
                  Object {
                    "IconComponent": Object {
                      "$$typeof": Symbol(react.memo),
                      "compare": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    "MenuProps": undefined,
                    "SelectDisplayProps": Object {
                      "id": undefined,
                    },
                    "autoWidth": false,
                    "children": Array [
                      <WithStyles(ForwardRef(MenuItem))
                        disabled={true}
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                        value="_default"
                      >
                        
                      </WithStyles(ForwardRef(MenuItem))>,
                      Array [
                        <WithStyles(ForwardRef(MenuItem))
                          value={true}
                        >
                          On
                        </WithStyles(ForwardRef(MenuItem))>,
                        <WithStyles(ForwardRef(MenuItem))
                          value={false}
                        >
                          Off
                        </WithStyles(ForwardRef(MenuItem))>,
                      ],
                    ],
                    "classes": Object {
                      "disabled": "Mui-disabled",
                      "filled": "MuiSelect-filled",
                      "icon": "MuiSelect-icon",
                      "iconFilled": "MuiSelect-iconFilled",
                      "iconOpen": "MuiSelect-iconOpen",
                      "iconOutlined": "MuiSelect-iconOutlined",
                      "nativeInput": "MuiSelect-nativeInput",
                      "outlined": "MuiSelect-outlined",
                      "root": "MuiSelect-root",
                      "select": "MuiSelect-select",
                      "selectMenu": "MuiSelect-selectMenu",
                    },
                    "displayEmpty": false,
                    "labelId": undefined,
                    "multiple": false,
                    "onClose": undefined,
                    "onOpen": undefined,
                    "open": undefined,
                    "renderValue": undefined,
                    "type": undefined,
                    "variant": "standard",
                  }
                }
                multiline={false}
                onChange={[Function]}
                onClick={[Function]}
                style={
                  Object {
                    "width": "100%",
                    "zIndex": 1,
                  }
                }
                type="text"
                value={true}
              >
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline"
                  onClick={[Function]}
                  style={
                    Object {
                      "width": "100%",
                      "zIndex": 1,
                    }
                  }
                >
                  <ForwardRef(SelectInput)
                    IconComponent={
                      Object {
                        "$$typeof": Symbol(react.memo),
                        "compare": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      }
                    }
                    SelectDisplayProps={
                      Object {
                        "id": undefined,
                      }
                    }
                    autoWidth={false}
                    className="MuiInputBase-input MuiInput-input"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "filled": "MuiSelect-filled",
                        "icon": "MuiSelect-icon",
                        "iconFilled": "MuiSelect-iconFilled",
                        "iconOpen": "MuiSelect-iconOpen",
                        "iconOutlined": "MuiSelect-iconOutlined",
                        "nativeInput": "MuiSelect-nativeInput",
                        "outlined": "MuiSelect-outlined",
                        "root": "MuiSelect-root",
                        "select": "MuiSelect-select",
                        "selectMenu": "MuiSelect-selectMenu",
                      }
                    }
                    disabled={false}
                    displayEmpty={false}
                    inputRef={[Function]}
                    multiple={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    value={true}
                    variant="standard"
                  >
                    <div
                      aria-haspopup="listbox"
                      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      On
                    </div>
                    <input
                      aria-hidden={true}
                      className="MuiSelect-nativeInput"
                      onAnimationStart={[Function]}
                      onChange={[Function]}
                      tabIndex={-1}
                      value={true}
                    />
                    <ForwardRef(ArrowDropDownIcon)
                      className="MuiSelect-icon"
                    >
                      <WithStyles(ForwardRef(SvgIcon))
                        className="MuiSelect-icon"
                      >
                        <ForwardRef(SvgIcon)
                          className="MuiSelect-icon"
                          classes={
                            Object {
                              "colorAction": "MuiSvgIcon-colorAction",
                              "colorDisabled": "MuiSvgIcon-colorDisabled",
                              "colorError": "MuiSvgIcon-colorError",
                              "colorPrimary": "MuiSvgIcon-colorPrimary",
                              "colorSecondary": "MuiSvgIcon-colorSecondary",
                              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                              "root": "MuiSvgIcon-root",
                            }
                          }
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSelect-icon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                        </ForwardRef(SvgIcon)>
                      </WithStyles(ForwardRef(SvgIcon))>
                    </ForwardRef(ArrowDropDownIcon)>
                    <WithStyles(ForwardRef(Menu))
                      MenuListProps={
                        Object {
                          "aria-labelledby": undefined,
                          "disableListWrap": true,
                          "role": "listbox",
                        }
                      }
                      PaperProps={
                        Object {
                          "style": Object {
                            "minWidth": undefined,
                          },
                        }
                      }
                      anchorEl={
                        <div
                          aria-haspopup="listbox"
                          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                          role="button"
                          tabindex="0"
                        >
                          On
                        </div>
                      }
                      id="menu-"
                      onClose={[Function]}
                      open={false}
                    >
                      <ForwardRef(Menu)
                        MenuListProps={
                          Object {
                            "aria-labelledby": undefined,
                            "disableListWrap": true,
                            "role": "listbox",
                          }
                        }
                        PaperProps={
                          Object {
                            "style": Object {
                              "minWidth": undefined,
                            },
                          }
                        }
                        anchorEl={
                          <div
                            aria-haspopup="listbox"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                            role="button"
                            tabindex="0"
                          >
                            On
                          </div>
                        }
                        classes={
                          Object {
                            "list": "MuiMenu-list",
                            "paper": "MuiMenu-paper",
                          }
                        }
                        id="menu-"
                        onClose={[Function]}
                        open={false}
                      >
                        <WithStyles(ForwardRef(Popover))
                          PaperProps={
                            Object {
                              "classes": Object {
                                "root": "MuiMenu-paper",
                              },
                              "style": Object {
                                "minWidth": undefined,
                              },
                            }
                          }
                          anchorEl={
                            <div
                              aria-haspopup="listbox"
                              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                              role="button"
                              tabindex="0"
                            >
                              On
                            </div>
                          }
                          anchorOrigin={
                            Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
                          }
                          getContentAnchorEl={[Function]}
                          id="menu-"
                          onClose={[Function]}
                          onEntering={[Function]}
                          open={false}
                          transformOrigin={
                            Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
                          }
                          transitionDuration="auto"
                        >
                          <ForwardRef(Popover)
                            PaperProps={
                              Object {
                                "classes": Object {
                                  "root": "MuiMenu-paper",
                                },
                                "style": Object {
                                  "minWidth": undefined,
                                },
                              }
                            }
                            anchorEl={
                              <div
                                aria-haspopup="listbox"
                                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                role="button"
                                tabindex="0"
                              >
                                On
                              </div>
                            }
                            anchorOrigin={
                              Object {
                                "horizontal": "left",
                                "vertical": "top",
                              }
                            }
                            classes={
                              Object {
                                "paper": "MuiPopover-paper",
                                "root": "MuiPopover-root",
                              }
                            }
                            getContentAnchorEl={[Function]}
                            id="menu-"
                            onClose={[Function]}
                            onEntering={[Function]}
                            open={false}
                            transformOrigin={
                              Object {
                                "horizontal": "left",
                                "vertical": "top",
                              }
                            }
                            transitionDuration="auto"
                          >
                            <ForwardRef(Modal)
                              BackdropProps={
                                Object {
                                  "invisible": true,
                                }
                              }
                              className="MuiPopover-root"
                              container={<body />}
                              id="menu-"
                              onClose={[Function]}
                              open={false}
                            />
                          </ForwardRef(Popover)>
                        </WithStyles(ForwardRef(Popover))>
                      </ForwardRef(Menu)>
                    </WithStyles(ForwardRef(Menu))>
                  </ForwardRef(SelectInput)>
                </div>
              </ForwardRef(InputBase)>
            </WithStyles(ForwardRef(InputBase))>
          </ForwardRef(Input)>
        </WithStyles(ForwardRef(Input))>
      </ForwardRef(Select)>
    </WithStyles(ForwardRef(Select))>
     
  </span>
</SelectInput>
`;

exports[`SelectInput Component render correctly when it is disabled 1`] = `
<span
  className=""
>
   
  <WithStyles(ForwardRef(Select))
    className=""
    disabled={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "width": "100%",
        "zIndex": 1,
      }
    }
    value={true}
  >
    <WithStyles(ForwardRef(MenuItem))
      disabled={true}
      style={
        Object {
          "display": "none",
        }
      }
      value="_default"
    />
    <WithStyles(ForwardRef(MenuItem))
      key="0"
      value={true}
    >
      On
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="1"
      value={false}
    >
      Off
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
   
</span>
`;
